src/shared/util.js

```diff
@@ -905,12 +905,21 @@ const PDFStringTranslateTable = [
 ];
 
 function stringToPDFString(str) {
+  // See section 7.9.2.2 Text String Type.
```

Comment:
Looking at https://opensource.adobe.com/dc-acrobat-sdk-docs/pdfstandards/PDF32000_2008.pdf#G6.1957385 it contains:
```
The escape sequence shall consist of the following elements, in order:
a) The Unicode value U+001B (that is, the byte sequence 0 followed by 27).
b) A 2- byte ISO 639 language code.
EXAMPLE en for English or ja for Japanese encoded as ASCII characters.
c) (Optional) A 2-byte ISO 3166 country code.
EXAMPLE US for the United States or JP for Japan.
d) The Unicode value U+001B.
```
With the changes below it looks like we'll basically skip *anything* after a `U+001B`, until a new `U+001B` is encountered, regardless of actual content between the delimiters?

I might be wrong, but technically it *seems* that a string looking e.g. something like `abc\x1bd` should then be left alone?
Most likely such a thing would be extremely rare, or even somewhat meaningless since the `\x1b` won't be visible anyway, but I can't help wondering if doing "more" than the specification says could break some weird edge-case?

(Perhaps I'm worrying to much here, as I often tend to do, but I just wanted to ask; this is likely fine as-is.)

Reply:
After a nights sleep, I do wonder if what the specification says doesn't mean that we should do the replacement on the final string using a regular expression such as the following?
```javascript
/\x1b[a-zA-Z]{2}(?:[a-zA-Z]{2})?\x1b/g
```

Reply:
I didn't manage to see how Acrobat is dealing with language codes (I tried with strings in Info or with ones in TU): they likely don't implement their own specs, but maybe I'm wrong.
I tested with Preview on Mac and it just strips out everything between the ESC markers or whatever after ESC if the closing one is missing.
And in few weeks, this behavior should the Chrome's one so I'd be inclined to do the same in Firefox.

Reply:
> I didn't manage to see how Acrobat is dealing with language codes (I tried with strings in Info or with ones in TU): they likely don't implement their own specs,

That's unfortunately not unlikely, since it wouldn't be the first time that's happened.

> I tested with Preview on Mac and it just strips out everything between the ESC markers or whatever after ESC if the closing one is missing.
> And in few weeks, this behavior should the Chrome's one so I'd be inclined to do the same in Firefox.

In that case I suppose that this should lower the overall risk, since any breakage would occur in many viewer. (And we could always refer back to this discussion, if any future tweaks are needed here.)